apps/DensifyPointCloud/DensifyPointCloud.cpp

```diff
@@ -296,6 +298,23 @@ int main(int argc, LPCTSTR* argv)
 	// load and estimate a dense point-cloud
 	if (!scene.Load(MAKE_PATH_SAFE(OPT::strInputFileName)))
 		return EXIT_FAILURE;
+	if (!OPT::strMaskPath.empty()) {
+		if (OPTDENSE::nIgnoreMaskLabel < 0) {
+			VERBOSE("--ignore-mask-label should be >=0 when --mask-path is set");
+			return EXIT_FAILURE;
+		}
+		for (Image& image : scene.images) {
+			if (!image.maskName.empty()) {
+				VERBOSE("Image %s has non-empty maskName %s", image.name.c_str(), image.maskName.c_str());
+				return EXIT_FAILURE;
+			}
+			image.maskName = Util::ensureFolderSlash(OPT::strMaskPath) + Util::getFileName(image.name) + ".mask.png";
```

Comment:
Util::ensureFolderSlash(OPT::strMaskPath) needs to be run once at the beginning only; and you should use ensureValidFolderPath instead, which does more checks